### PR TITLE
DDF-2547 Updated URLResourceReader to use basic auth username/password if present.

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Updates the URLResourceReader to handle basic auth username/password
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@gordocanchola @brianfelix 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).

[Security](https://github.com/orgs/codice/teams/security)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@shaundmorris
@stustison
#### How should this be tested? (List steps with links to updated documentation)
Verify build passes
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2547
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/core/catalog-core-urlresourcereader/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 1]
```diff
@@ -30,7 +30,20 @@
...
+        <cm:managed-properties
+                persistent-id="ddf.catalog.resource.impl.URLResourceReader"
+                update-strategy="container-managed"/>
+        <argument ref="mimeTypeMapper"/>
+        <property name="rootResourceDirectories">
+            <set>
+                <value type="java.lang.String">${ddf.home}/data/products</value>
+            </set>
+        </property>
+        <property name="followRedirects" value="true"/>
```
Comment (stustison) [inline catalog/core/catalog-core-urlresourcereader/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: We shouldn't be following redirects by default.

Comment (clockard) [inline catalog/core/catalog-core-urlresourcereader/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: This is that secondary instance we created specifically for following redirects. It is exposed with a different id and shortname. This is what was needed to support things like the confluence source.

**Comment by shaundmorris** [comment]
Should we add a unit test to use the username and password? Not sure how valuable that would be.

**Comment by clockard** [comment]
@shaundmorris I looked at creating a unit test for the username and password but I couldn't see a way to pull that info back out once it was set.

**Comment by clockard** [comment]
Successfully built and tested :shipit:

**Pull request merged**